src/rwe/grid/DiscreteRect.h

```diff
@@ -13,11 +13,11 @@ namespace rwe
 
         int x;
         int y;
-        unsigned int width;
-        unsigned int height;
+        int width;
+        int height;
 
         DiscreteRect() = default;
-        DiscreteRect(int x, int y, unsigned int width, unsigned int height) : x(x), y(y), width(width), height(height)
+        DiscreteRect(int x, int y, int width, int height) : x(x), y(y), width(width), height(height)
```

Comment:
I wonder if the constructor should assert or throw if `width` and `height` are <0 now that it is possible.

Reply:
Just brainstorming... So if it is possible they are <0 now, it was possible before that arithmetic errors(?) before would've produced enormous unsigned width/height values. Those _also_ would be bugs, but uncaught. So I guess it could be argued that yes, we should put in an assert, and that is an improvement vs the previous unsigned code. Thoughts @MHeasell ?